org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/quickfix/ITextModification.java

```diff
@@ -0,0 +1,19 @@
+/*******************************************************************************
+ * Copyright (c) 2022 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.xtext.ide.editor.quickfix;
+
+/**
+ * @author Rub�n Porras Campo - Initial contribution and API
+ *
+ * @since 2.27
+ */
+@FunctionalInterface
+public interface ITextModification<A, B, C, R> {
+  R apply(A a, B b, C c);
```

Comment:
Don't think any code uses two space as indent

Reply:
What are a, b and c? Might be useful to give them better names or document them

Reply:
Also I don't feel like the current ITextModification looks like an ITextModification at all. More like a Function3.

Judging from the usages, the parameterizations are the same for all. Maybe remove the unnecessary type parameters.

Reply:
Done, i have removed the type parameters